rpm/archivematica-storage-service/package.spec

```diff
@@ -76,12 +76,17 @@ rm -rf %{buildroot}
 
 %post
 
-# Create archivematica user
-userID=`id -u archivematica`
-if [ "${userID}" != 333 ]; then
-  useradd --uid 333 --user-group --home /var/lib/archivematica/ archivematica
+# Create archivematica user and group
+getent group archivematica >/dev/null || groupadd -f -g 333 -r archivematica
+if ! getent passwd archivematica >/dev/null ; then
+    if ! getent passwd 333 >/dev/null ; then
+      useradd -r -u 333 -g archivematica -d /var/lib/archivematica/ -s /sbin/nologin -c "Archivematica system account" archivematica
+    else
+      useradd -r -g archivematica -d /var/lib/archivematica/ -s /sbin/nologin -c "Archivematica system account" archivematica
+    fi
 fi
```

Comment:
The indentation in the block above is inconsistent, it mixes 2 (L83 and L85) and 4 whitespaces.

Reply:
Should we use variables to make this a bit clearer?
HOMEDIR, USERNAME, GROUPNAME, ALLOCATED_UID and ALLOCATED_GID?